auto-value-cursor/src/main/java/com/gabrielittner/auto/value/cursor/AutoValueCursorExtension.java

```diff
@@ -124,21 +111,25 @@ private MethodSpec createReadMethod(Context context, String className,
                 .returns(returnType)
                 .addParameter(CURSOR, "cursor");
 
-        String[] propertyNames = new String[properties.keySet().size()];
-        propertyNames = properties.keySet().toArray(propertyNames);
+        Set<String> keySet = properties.keySet();
+        String[] propertyNames = new String[keySet.size()];
+        propertyNames = keySet.toArray(propertyNames);
         for (String name : propertyNames) {
             ExecutableElement element = properties.get(name);
             TypeName type = TypeName.get(element.getReturnType());
-
             String cursorMethod = getCursorMethod(type);
             String columnName = getColumnName(element);
+            TypeMirror factoryTypeMirror = getFactoryTypeMirror(element);
             if (cursorMethod != null) {
                 readMethod.addStatement(cursorMethod, type, name,
                         columnName != null ? columnName : name);
+            } else if (factoryTypeMirror != null) {
+                TypeName factoryType = TypeName.get(factoryTypeMirror);
+                readMethod.addStatement("$T $N = $T.create(cursor)", type, name, factoryType);
```

Comment:
It's nicer to stay flexible on the naming of the factory method. The factory could just be another auto value class and the extension only looks at return type and parameter in `applicable()`. 

As an example:
```java
@AutoValue public abstract class User {
  abstract String id();
  abstract String name();
  @AutoValueCursorFactory(Avatar.class) Avatar avatar();

  public static User create(Cursor cursor) {
    return AutoValue_User.createFromCursor(cursor);
  }
}

@AutoValue public abstract class Avatar {
  public abstract String smallImageUrl();
  public abstract String largeImageUrl();

  public Avatar fromCursor(Cursor cursor) {
    return AutoValue_Avatar.createFromCursor(cursor);
  }
}
```